### ISSUE TITLE
Report background and highlighting colors in Microsoft Word

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#5866]
I noticed that NVDA is unable to report background and highlighting color changes in Microsoft Word. Check [the attachment](https://github.com/nvaccess/nvda/files/208986/Color.reporting.zip) for an example.

Steps to reproduce
1. Type a two lines of text in an empty document
2. Select the second line of text
3. Press alt+h,i and select a highlighting color
4. Select again, press alt+h,h and select a differenc background color
5. Arrow up and down in the document with color reporting enabled

Actual result: NVDA reports no difference
Proposed result: NVDA reports both background and highlighting color changes

**Comment by leonardder** [comment]
Given the recent work that has been done to improve MS Word support (#2961, #5800, #5946), I'd like to bring this ticket under attention again.

@feerrenrut, mind assigning a priority to this?

**Comment by nishimotz** [comment]
Requested the enhancement in Japan, regarding Microsoft Outlook as well.

**Comment by Adriani90** [comment]
This would be really helpful. I hope it will come any time in the future into NVDA.

**Comment by ABuffEr** [comment]
Hi @leonardder ,

I heavily played with UIA and VBA in recent days, around this situation.

Imho, it's very confused even API side.

Had you find a reliable way to distinguish between background and highlight color?

I found only [this property](https://learn.microsoft.com/en-us/office/vba/api/Word.Range.HighlightColorIndex) useful to be sure whether there is an highlighting and what's the color applied...

**Comment by leonardder** [comment]
I haven't played with this for years, so I have no idea honestly.

**Comment by ABuffEr** [comment]
Meanwhile I encountered problems with attached document, so i suspect the situation is changed from 2016, even in current Word support of these formatting in documents made with previous versions.

However, Range.[Shading.BackgroundPatternColor](https://learn.microsoft.com/en-us/office/vba/api/Word.Shading.BackgroundPatternColor) seems a good candidate to query for background.

I'll continue to investigate, these seem to be things quite important in school, for teachers and students.
